### PR TITLE
Resurrect BrowserSelectorTest

### PR DESCRIPTION
**Pull request by rpdome**
This is breaking one of the pr pipeline. (The reason is that Mockito-android cannot mock final class).

I'm switching them to Robolectric.

Note: addResolveInfoForIntent() is already deprecated, but I can't find how to get the alternative work (since it requires ComponentName, ComponentName requires class name and we're not providing one here). I didn't want to spend too much time on it so I just pick the deprecated one for now.

### CODE REVIEW

File: common/src/test/java/com/microsoft/identity/common/BrowserSelectorTest.java [hunk 1]
```diff
@@ -253,14 +225,27 @@ public TestBrowserBuilder setVersion(String version) {
...
+        private PackageInfo addSignatures(final PackageInfo packageInfo, final Signature[] signatures) {
+            if (Build.VERSION.SDK_INT < Build.VERSION_CODES.P) {
+                packageInfo.signatures = signatures;
+                return packageInfo;
+            }
+
+            final SigningInfo signingInfo = mock(SigningInfo.class);
+            when(signingInfo.hasMultipleSigners()).thenReturn(false);
+            when(signingInfo.getSigningCertificateHistory()).thenReturn(signatures);
+            packageInfo.signingInfo = signingInfo;
```
Comment (shahzaibj) [inline common/src/test/java/com/microsoft/identity/common/BrowserSelectorTest.java:237]: Can this be done with Robolectric too?

File: common/src/test/java/com/microsoft/identity/common/BrowserSelectorTest.java [hunk 2]
```diff
@@ -253,14 +225,27 @@ public TestBrowserBuilder setVersion(String version) {
...
+        }
+
         public TestBrowser build() {
             Signature[] signatures = new Signature[mSignatures.size()];
 
             for (int i = 0; i < mSignatures.size(); i++) {
                 signatures[i] = new Signature(mSignatures.get(i));
             }
 
-            final PackageInfo pi = Util.addSignatures(new PackageInfo(), signatures);
```
Comment (shahzaibj) [inline common/src/test/java/com/microsoft/identity/common/BrowserSelectorTest.java:263]: Is the Util method still there? Can we delete it if there are no usages?

**Review by shahzaibj — APPROVED** [review]
just minor comments, else LGTM!

File: common/src/test/java/com/microsoft/identity/common/BrowserSelectorTest.java (hunk 1 above)
Comment (AdamBJohnsonx) [reply common/src/test/java/com/microsoft/identity/common/BrowserSelectorTest.java:237]: Don't know that the methodology matters, but when you're writing code, if you want to make it easy to test - implement an interface, and provide a way to construct your POJOs.

Also, at some point, SigningInfo has been made final.

**Review by shoatman — APPROVED** [review]

**Review by AdamBJohnsonx — APPROVED** [review]
I don't see an issue here.

File: common/src/test/java/com/microsoft/identity/common/BrowserSelectorTest.java (hunk 2 above)
Comment (rpdome) [reply common/src/test/java/com/microsoft/identity/common/BrowserSelectorTest.java:263]: It's still used.

File: common/src/test/java/com/microsoft/identity/common/BrowserSelectorTest.java (hunk 1 above)
Comment (rpdome) [reply common/src/test/java/com/microsoft/identity/common/BrowserSelectorTest.java:237]: Spent a few mins with it and couldn't think of a way of doing this cleanly with robolectric (this is 'per browser', and there are >1 browsers in each tests), so I'm keeping this as is for now.

**Pull request merged**